### PR TITLE
HiPACE++ 23.07

### PR DESCRIPTION
**Pull request by MaxThevenet**
Update the package, in particular:
- Add recent versions 23.07 and 23.05 (so far only 21.09 is available)
- Update minimal CMake version
- Update minimal openPMD-api version
- Enforce compatible GCC/CUDA versions to build HiPACE++

### CODE REVIEW

**Review by tldahlgren — DISMISSED** [review]
Confirmed the new version sha256.  Deferring merge to give @ax3l  and or @SeverinDiederichs time to comment.

File: var/spack/repos/builtin/packages/hipace/package.py [hunk 1]
```diff
@@ -37,10 +39,11 @@ class Hipace(CMakePackage):
         description="Floating point precision (single/double)",
     )
 
+    depends_on("cmake@3.18.0:", type="build", when="@23.07:")
     depends_on("cmake@3.15.0:", type="build")
-    depends_on("cuda@9.2.88:", when="compute=cuda")
     depends_on("mpi", when="+mpi")
     with when("+openpmd"):
+        depends_on("openpmd-api@0.15.1:", when="@23.07:")
```
Comment (MaxThevenet) [inline var/spack/repos/builtin/packages/hipace/package.py]: Pointed out by @SeverinDiederichs: v23.05 already requires the latest openPMD-api.
```suggestion
        depends_on("openpmd-api@0.15.1:", when="@23.05:")
```

File: var/spack/repos/builtin/packages/hipace/package.py [hunk 2]
```diff
@@ -37,10 +39,11 @@ class Hipace(CMakePackage):
         description="Floating point precision (single/double)",
     )
 
+    depends_on("cmake@3.18.0:", type="build", when="@23.07:")
```
Comment (MaxThevenet) [inline var/spack/repos/builtin/packages/hipace/package.py]:
```suggestion
    depends_on("cmake@3.18.0:", type="build", when="@23.05:")
```

**Review by SeverinDiederichs — APPROVED** [review]
Thank you for the update. I tested with `compute=cuda` and confirm that it works.

**Review by SeverinDiederichs — APPROVED** [review]

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**